### ISSUE TITLE
Size of metabase.db.mv.db increase from megabytes to gigabytes

### ISSUE THREAD

**Issue by anki-code** [metabase/metabase#6034]
Metabase 0.25.2 FYI @salsakran @tlrobinson 

🔥 One day I've seen that size of `metabase.db.mv.db` increase from megabytes to gigabytes.

I reviewed the tables:
```
$ wget http://repo2.maven.org/maven2/com/h2database/h2/1.4.196/h2-1.4.196.jar
$ java -cp h2*.jar org.h2.tools.Shell -driver org.h2.Driver -url "jdbc:h2:./metabase.db" -user "" -password ""
sql> SELECT count(*) FROM METABASE_FIELDS;
```
I've seen that the Metabase automatically adds hundreds of thousands records to the METABASE_FIELD table. After review the tables I've found wrong collection in MongoDB where the key was ID and Metabase loads the ID as column name to `metabase.db.mv.db`.

🚒 My first suggestion is to add the check the count of columns in table and don't upload probably wrong tables to the Metabase DB.

Next step I remove the MongoDB database with wrong collection for test. I've seen that size of `metabase.db.mv.db` was reduced twice but it stays gigabytes and not becomes megabytes. I think it's because indexes for H2 tables or something like this aren't rebuild.

🚒 The second suggestion is to check the index rebuild process.

Next step I rebuild the Metabase database by import the database into SQL file and build new H2 database:
```
$ docker metabase stop
$ java -cp h2*.jar org.h2.tools.Script -url "jdbc:h2:./metabase.db" -user "" -password "" -script metabase-dump.sql
$ java -cp h2*.jar org.h2.tools.RunScript -url jdbc:h2:./metabase-new.db -script metabase-dump.sql
$ mv metabase.db.mv.db metabase-old.db.mv.db; 
$ mv metabase-new.db.mv.db metabase.db.mv.db
$ docker metabase start
```

After it I get the good size of `metabase.db.mv.db` around megabytes. Success!

🚒 The third suggestion is to add the check of the database size (show warnings) AND the info about database state in the Admin panel of Metabase

⬇️ Click "+1" icon instead of "+1" comment

**Comment by salsakran** [comment]
"My first suggestion is to add the check the count of columns in table and don't upload probably wrong tables to the Metabase DB." 

Any thoughts on the logic for this? It's hard to predict all the weird shapes of data people have, but if you have concrete ideas here, we're happy to try to figure something out.

**Comment by anki-code** [comment]
In my case Metabase inserted hundreds of thousands columns into a METABASE FIELD table. 
I think the behavior could be: if the detected count of columns more than 100 or 1000, don't sync the table by default and show the warning and button "Switсh the sync on" in the "Admin → Data Model → Table name" section.

**Comment by roemhildtg** [comment]
Any easy workarounds for this? I just noticed our db size grow to 11GB

**Comment by flamber** [comment]
Hi @roemhildtg 

If you're using 0.32, it should now do some cleanup, when Metabase is stopping (#9411), because that process cannot be done while Metabase is using the database.

Please do not use the H2 database in production - migrate to Postgres/MySQL:
https://metabase.com/docs/latest/operations-guide/start.html#migrating-from-using-the-h2-database-to-mysql-or-postgres

You can manually run cleanup on H2 - have a look in this thread:
https://discourse.metabase.com/t/metabase-db-cleanup/5079?u=flamber

**Comment by roemhildtg** [comment]
Okay - good to know. We'll work on migrating to postgres or mysql. Thanks  @flamber!

**Comment by camsaul** [comment]
Fixed by #9411
